enhancements/update/duration-of-migration-to-multi-arch.md

```diff
@@ -0,0 +1,167 @@
+The version of MicroShift is managed by [dnf](https://docs.redhat.com/fr/documentation/red_hat_build_of_microshift/4.17/html-single/installing_with_an_rpm_package/index#microshift-install-rpm), not CVO and thus MicroShift does not hit the issue described here.
+
+### Implementation Details/Notes/Constraints
+
+The changes in this section are also reflected in [clusteroperator.md](../../dev-guide/cluster-version-operator/dev/clusteroperator.md).
+
+Each operator creates a pull request that includes:
+
+- in the `ClusterVersion` manifest:
+
+  - a new element with `name=operator-image` and `version=placeholder.url.oc.will.replace.this.org/placeholdernamespace:<operator-tag-name>` where `<operator-tag-name>` is the name of the tag for the operator in [image-references](./operators.md#how-do-i-ensure-the-right-images-get-used-by-my-manifests).
+  - OPTIONALLY a new element with `name=operand-image` and `version=placeholder.url.oc.will.replace.this.org/placeholdernamespace:<operand-tag-name>` where `<operand-tag-name>` is the name of the tag for an operator's operand in [image-references](./operators.md#how-do-i-ensure-the-right-images-get-used-by-my-manifests).
+
+
+- in the `deployment` manifest, environment variables or flags with a placeholder for an image pull specification that will be replaced when the payload is built.
+
+- in the operator's implementation, the operator takes the values of the environment variables or the flags and use them to populate the `clusteroperatror.status.versions`.
+
+A special operator: [machine-config-operator](https://github.com/openshift/machine-config-operator) (MCO).
+It is special because it is the last operator to upgrade in a cluster upgrade and it usually takes longest for MCO to upgrade. With the new populated elements in `clusteroperator.status.versions`, the ending point of migrating to multi-arch will become accurate which improves the overall UX the most.
+The pull request [MCO#4637](https://github.com/openshift/machine-config-operator/pull/4637) is created for the purpose.
+Other operators may take the pull request as an example when it is ready to report the image pull specifications in the operator's `.status.versions`.
+
+### Risks and Mitigations
+
+As we allow operators to modify their implementation on its own pace. The upgrade progress of migrating to multi-arch only gets better when more operators take the actions. No risks are foreseen at the moment.
+
+### Drawbacks
+
+All OpenShift cluster operators have to modify their implementation to make the whole migration process accurate.
+
+## Open Questions [optional]
+
+## Test Plan
+
+An e2e test which is executed in an optional pre-submit at the beginning until CVO and an operator (or MCO?) adopt the new API, exists for CVO that monitors the upgrade to multi-arch. It passes only when the expected `clusterversion/version`, events or logs with information about the migration are observed. This might require the modification on CVO.
```

Comment:
one test plan would be:

* Install a single-arch release
* Request an update to a multi-arch release.
* Ensure that update completes (ClusterVersion `status.conditions` having `Progressing=False` and `status.history` claiming `Completed`).
* Immediately after ClusterVersion claims completeness, request a cross-arch compute instance (Machine(Set) for standalone, NodePool for HyperShift).
* Ensure that that cross-arch compute instance successfully joins the cluster as a Node, within some reasonable time threshold (15m?).

Reply:
Updated.